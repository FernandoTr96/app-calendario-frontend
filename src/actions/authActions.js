import { fetchWithoutToken, fetchWithToken } from "../helpers/fetch";
import { types } from "../types/types";
import { toast } from 'react-toastify';

export const startLoginWithEmailAndPassword = (email, password)=>{

    return async (dispatch)=>{

        const response = await fetchWithoutToken(
            'auth',
            {email, password},
            'POST'
        );

        const body = await response.json();
        
        if(body.ok){
            localStorage.setItem('calendarAppToken', body.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            delete body.token;
            delete body.ok;
            dispatch(login(body));
        }
        else{

            toast.error('usuario o contraseÃ±a incorrecta', {
                position: "top-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

    }

}

export const login = (userCredentials)=>({
    type: types.login,
    payload: userCredentials
})

export const registerAccount = (userdata)=>{
    
    return async (dispatch)=>{

        const response = await fetchWithoutToken(
            'auth/register',
            {...userdata},
            'POST'
        );

        const body = await response.json();

        if(body.ok){
            localStorage.setItem('calendarAppToken', body.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            delete body.token;
            delete body.ok;
            delete body.confirm_password;
            dispatch(login(body));
        }
        else{

            toast.error(body.msg, {
                position: "top-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        if(body.errors?.length > 0){
            body.errors.forEach(field => {

                toast.error(field.msg, {
                    position: "top-right",
                    autoClose: 5000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });

            })
        }

    }

}

export const startChecking = ()=> {

    return async (dispatch)=>{

        const response = await fetchWithToken(
            'auth/revalidate-token',
            'GET'
        );

        const body = await response.json();

        if(body.ok){
            localStorage.setItem('calendarAppToken', body.token );
            localStorage.setItem('token-init-date', new Date().getTime() );
            delete body.token;
            delete body.ok;
            dispatch(login(body));
        }
        else{
            dispatch(checkingFinish());
        }

    }

}

export const checkingFinish = ()=>({
    type: types.checkingFinish
})

export const startLogout = ()=>{

    return (dispatch)=>{

        localStorage.clear();
        dispatch(logout());

    }

}

export const logout = ()=>({
    type: types.purgeAuth
})