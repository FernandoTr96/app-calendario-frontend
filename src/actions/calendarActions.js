import { types } from "../types/types";
import {fetchWithToken} from '../helpers/fetch'
import {toast} from 'react-toastify'
import { string2date } from "../helpers/string2date";

export const getEvents = ()=>{
    return async (dispatch)=>{

        dispatch(showLoader());

        try {

            const response = await fetchWithToken('events');
            const body = await response.json();
            

            if(response.ok){
                
                const events = string2date(body.events);
                dispatch(setEvents(events));

            }

            
        } catch (error) {

            toast.error(`Error : ${error}`, {
                position: "bottom-right",
                autoClose: 4000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        dispatch(hiddeLoader());

    }
}

export const setEvents = (events) =>
(
    {
        type: types.fillEventList,
        payload: events
    }
)

export const saveEvent = (event)=>{
    return async (dispatch) => {

        dispatch(showLoader());

        try {

            const insert = await fetchWithToken(
                'events/addEvent',
                event,
                'POST'
            );

            if(insert.ok){
                toast.success(`Se ha aÃ±adido un nuevo evento al calendario`, {
                    position: "bottom-right",
                    autoClose: 3000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });
                dispatch(getEvents());
            }
            
        } catch (error) {
            
            toast.error(error.msg, {
                position: "bottom-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        dispatch(hiddeLoader());

    }
}


export const editEvent = (event)=>{
    return async (dispatch)=>{

        const id = event.id;
        delete event.id;

        dispatch(showLoader());

        try {

            const update =  await fetchWithToken(
                `events/update/${id}`,
                event,
                'PUT'
            );

            if(update.ok){

                toast.info(`evento actualizado con exito`, {
                    position: "bottom-right",
                    autoClose: 3000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });

                dispatch(getEvents());

            }

        } catch (error) {
            
            toast.error(`Error: ${error.msg}`, {
                position: "bottom-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });
        }

        dispatch(hiddeLoader());

    }
}

export const deleteEvent = (id)=>{
    return async (dispatch)=>{

        dispatch(showLoader());

        try {

            const _delete = await fetchWithToken(
                `events/delete/${id}`,
                {},
                'DELETE'
            )

            if(_delete.ok){

                toast.warning(`el evento fue eliminado con exito`, {
                    position: "bottom-right",
                    autoClose: 3000,
                    hideProgressBar: false,
                    closeOnClick: true,
                    pauseOnHover: false,
                    draggable: false,
                    progress: undefined,
                });

                dispatch(getEvents());
            }

            
        } catch (error) {

            toast.error(`Error: ${error.msg}`, {
                position: "bottom-right",
                autoClose: 5000,
                hideProgressBar: false,
                closeOnClick: true,
                pauseOnHover: false,
                draggable: false,
                progress: undefined,
            });

        }

        dispatch(hiddeLoader());

    }
}

export const activeEvent = (event)=>(
    {
        type: types.activeEvent,
        payload: event
    }
)

export const showLoader = ()=>(
    {
        type: types.showLoader
    }
)
export const hiddeLoader = ()=>(
    {
        type: types.hiddeLoader
    }
)