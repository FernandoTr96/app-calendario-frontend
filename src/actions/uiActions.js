import { types } from "../types/types";

export const openModalEvents = ()=>(
    {
        type: types.uiOpenModalEvents,
        payload: true
    }
);

export const closeModalEvents = ()=>(
    {
        type: types.uiCloseModalEvents,
        payload: false
    }
);