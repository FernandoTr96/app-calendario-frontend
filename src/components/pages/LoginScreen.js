import React from 'react'
import { Box } from '@mui/material'
import styles from '../../styles/mui-sx-styles/LoginScreenStyles'
import { SliderControls } from '../auth/SliderControls'
import { LoginForm } from '../auth/LoginForm'
import { RegisterForm } from '../auth/RegisterForm'
import { ToastContainer } from 'react-toastify';

export const LoginScreen = () => {
    return (

        // main container
        <Box component='main' sx={styles.mainContainer} >

            {/* auth container */}
            <Box component='div' sx={styles.authContainer} >

                {/* tabs */}
                <SliderControls />

                {/* slider */}
                <Box component='div' sx={styles.slider} className='slider'>

                    {/* LOGIN */}
                    <LoginForm />

                    {/* REGISTER */}
                    <RegisterForm />

                </Box>

            </Box>

            <ToastContainer
                position="top-right"
                autoClose={5000}
                hideProgressBar={false}
                newestOnTop={false}
                closeOnClick
                rtl={false}
                pauseOnFocusLoss={false}
                draggable={false}
                pauseOnHover={false}
            />

        </Box>

    )
}
