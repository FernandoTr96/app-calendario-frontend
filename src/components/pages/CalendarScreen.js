import React, { useEffect } from 'react'
import { NavBar } from '../ui/NavBar'
import { Box } from '@mui/material'

// importaciones de big calendar 
// https://github.com/jquense/react-big-calendar#readme// 
import { Calendar, momentLocalizer } from 'react-big-calendar'
import moment from 'moment'
import 'moment/locale/es-mx';
import {messages} from '../../helpers/calendar-messages'
import 'react-big-calendar/lib/css/react-big-calendar.css'
import { theme } from '../../styles/mui-core-theme/materialui'
import { ModalEvents } from '../ui/ModalEvents'
import { useSelector, useDispatch } from 'react-redux'
import { activeEvent, getEvents } from '../../actions/calendarActions'
import { ToastContainer } from 'react-toastify';
import Stack from '@mui/material/Stack';
import LinearProgress from '@mui/material/LinearProgress';


export const CalendarScreen = () => {

    const dispatch = useDispatch();
    const {eventList, loader} = useSelector(state => state.events);
    const {uid} = useSelector(state => state.auth.account);

    const localizer = momentLocalizer(moment);
    const eventStyleGetter = (event /*, start, end, isSelected*/)=>{
        
        const style = {
            backgroundColor: event.user._id === uid ? theme.palette.secondary.light : theme.palette.secondary.dark,
            borderRadius: 'none',
            display: 'block',
            color: 'wintersmoke',
            outlineColor: '#333'
        }

        return{
            style
        }

    };

    const onSelectEvent = (eventObj) =>{
        dispatch(activeEvent(eventObj));
    }

    const onSelectSlot = ()=>{
        dispatch(activeEvent(null));
    }

    useEffect(() => {

        dispatch(getEvents());

    }, [dispatch])

    return (
        <Box component='main' className='calendar-screen'>
            
            {
                loader && 
                <Stack sx={{ width: '100%', color: 'grey.500' }} spacing={2}>
                    <LinearProgress color="warning" />
                </Stack>
            }

            <NavBar />

            <Calendar
                localizer={localizer}
                events={eventList}
                startAccessor='start'
                endAccessor='end'
                messages={messages}
                eventPropGetter={eventStyleGetter}

                onSelectEvent={onSelectEvent}
                onSelectSlot={onSelectSlot}
                selectable={true}
            />

            <ModalEvents/>

            <ToastContainer
                position="top-right"
                autoClose={5000}
                hideProgressBar={false}
                newestOnTop={false}
                closeOnClick
                rtl={false}
                pauseOnFocusLoss={false}
                draggable={false}
                pauseOnHover={false}
            />
        </Box>
    )
}
