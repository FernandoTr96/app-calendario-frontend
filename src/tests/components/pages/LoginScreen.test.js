import { mount } from 'enzyme'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'
import configure  from 'redux-mock-store'
import thunk from 'redux-thunk'
import { act } from 'react-dom/test-utils';
import { LoginScreen } from '../../../components/pages/LoginScreen';
import {startLoginWithEmailAndPassword} from '../../../actions/authActions';

jest.mock('../../../actions/authActions', ()=>({
    startLoginWithEmailAndPassword: jest.fn(),
    registerAccount: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configure(middlewares);
const initState = {};
let store = mockStore(initState);
store.dispatch = jest.fn();

let wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <LoginScreen/>
        </MemoryRouter>
    </Provider>
);

describe('Pruebas en <LoginScreen/>', () => {
    
    beforeEach(()=>{
        wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <LoginScreen/>
                </MemoryRouter>
            </Provider>
        );
    })

    test('debe renderizarse correctamente', () => {
    
        expect(wrapper).toMatchSnapshot();

    });

    test('debe llamar el dispatch del login ', async () => {

        const login = wrapper.find('form').at(0);
        
        await act( async ()=>{

            login.find('input[name="email"]').simulate('change', {
                target:{
                    name: 'email',
                    value: 'test@gmail.com'
                }
            })

            login.find('input[name="password"]').simulate('change', {
                target:{
                    name: 'password',
                    value: '123'
                }
            })

            login.simulate('submit', {
                preventDefault(){}
            })

        })

        expect(startLoginWithEmailAndPassword).toHaveBeenCalled();

    });

    test('debe mostrar mensaje de error cuando las contraseñas no son iguales en el registro', async () => {

        const register = wrapper.find('form').at(1);
        
        await act( async ()=>{

            register.find('input[name="name"]').simulate('change', {
                target:{
                    name: 'name',
                    value: 'fernando'
                }
            })

            register.find('input[name="email"]').simulate('change', {
                target:{
                    name: 'email',
                    value: 'test@gmail.com'
                }
            })

            register.find('input[name="password"]').simulate('change', {
                target:{
                    name: 'password',
                    value: '54321haloO'
                }
            })

            register.find('input[name="confirm_password"]').simulate('change', {
                target:{
                    name: 'confirm_password',
                    value: '54321haloOO'
                }
            })

            register.simulate('submit', {
                preventDefault(){}
            })

        })

        expect(register.find('.MuiFormControl-root').at(7).text().includes('Las contraseñas no coinciden')).toBe(true)

    });

});