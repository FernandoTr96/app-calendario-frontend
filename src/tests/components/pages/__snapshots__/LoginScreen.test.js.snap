// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen/> debe renderizarse correctamente 1`] = `
<main
  className="MuiBox-root css-1jhmcd4"
>
  <div
    className="MuiBox-root css-136gdm6"
  >
    Array [
      <input
        defaultChecked={true}
        id="tabLogin"
        name="tab"
        type="radio"
      />,
      <input
        id="tabRegister"
        name="tab"
        type="radio"
      />,
      <div
        className="MuiBox-root css-1g1bc7k"
      >
        <label
          className="tabLogin MuiBox-root css-0"
          htmlFor="tabLogin"
        >
          LOGIN
        </label>
        <label
          className="tabRegister MuiBox-root css-0"
          htmlFor="tabRegister"
        >
          REGISTER
        </label>
        <span
          className="bar MuiBox-root css-0"
        />
      </div>,
    ]
    <div
      className="slider MuiBox-root css-1jn54zx"
    >
      <form
        autoComplete="off"
        className="MuiBox-root css-0"
        onSubmit={[Function]}
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-11kybd2-MuiTypography-root"
        >
          Calendar app
        </h4>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          Array [
            "",
            <div
              className="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-4wkuxf-MuiInputBase-root-MuiFilledInput-root"
              onClick={[Function]}
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
              >
                <span
                  className="notranslate"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="AlternateEmailIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10h5v-2h-5c-4.34 0-8-3.66-8-8s3.66-8 8-8 8 3.66 8 8v1.43c0 .79-.71 1.57-1.5 1.57s-1.5-.78-1.5-1.57V12c0-2.76-2.24-5-5-5s-5 2.24-5 5 2.24 5 5 5c1.38 0 2.64-.56 3.54-1.47.65.89 1.77 1.47 2.96 1.47 1.97 0 3.5-1.6 3.5-3.57V12c0-5.52-4.48-10-10-10zm0 13c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                    />
                  </svg>
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
                disabled={false}
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="correo"
                required={false}
                type="text"
              />
            </div>,
          ]
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          Array [
            "",
            <div
              className="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-4wkuxf-MuiInputBase-root-MuiFilledInput-root"
              onClick={[Function]}
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium css-9mc7sf-MuiInputAdornment-root"
              >
                <span
                  className="notranslate"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="LockOpenIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm6-9h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6h1.9c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm0 12H6V10h12v10z"
                    />
                  </svg>
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
                disabled={false}
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="contraseÃ±a"
                required={false}
                type="password"
              />
            </div>,
          ]
        </div>
        <button
          className="MuiButton-root MuiButton-gradient-citrus MuiButton-gradient-citrusPrimary MuiButton-sizeMedium MuiButton-gradient-citrusSizeMedium MuiButton-fullWidth MuiButtonBase-root css-jt6ki8-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          iniciar sesion
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
      <form
        autoComplete="off"
        className="MuiBox-root css-0"
        onSubmit={[Function]}
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 css-11kybd2-MuiTypography-root"
        >
          Crea una cuenta
        </h4>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            nombre
          </label>
          Array [
            "",
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                name="name"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    nombre
                  </span>
                </legend>
              </fieldset>
            </div>,
          ]
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            correo
          </label>
          Array [
            "",
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    correo
                  </span>
                </legend>
              </fieldset>
            </div>,
          ]
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            contraseÃ±a
          </label>
          Array [
            "",
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                    />
                  </svg>
                  <span
                    className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    contraseÃ±a
                  </span>
                </legend>
              </fieldset>
            </div>,
          ]
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
          >
            confirmar constraseÃ±a
          </label>
          Array [
            "",
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                name="confirm_password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="password"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-dhh0px"
                >
                  <span>
                    confirmar constraseÃ±a
                  </span>
                </legend>
              </fieldset>
            </div>,
          ]
        </div>
        <button
          className="MuiButton-root MuiButton-gradient-citrus MuiButton-gradient-citrusPrimary MuiButton-sizeMedium MuiButton-gradient-citrusSizeMedium MuiButton-fullWidth MuiButtonBase-root css-jt6ki8-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Registrarse
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
  <div
    className="Toastify"
  />
</main>
`;
